source/geod24/concurrency.d

```diff
@@ -1101,6 +1103,11 @@ private:
     {
         import std.algorithm.mutation : remove;
 
+        if (this.dispatching)
+            return;
+
+        this.dispatching = true;
```

Comment:
Where did you have the problem ? `dispatch` is a `private` function only called from the constructor, so it should never be called twice to begin with.

Reply:
Sometimes called twice for FiberScheduler.start.

Reply:
Before one enters dispatch, if FibreScheduler.start is called...

Reply:
`start` should only every be called once though. Calling it twice is a programmer's error. So the check should probably be there (and be an assert failure).